### PR TITLE
usm: Re-enable Kafka raw split tests with workaround in proxy

### PR DESCRIPTION
**Pull request by vitkyrka**
### What does this PR do?
   
Occasional errors have been seen in the Kafka TLS raw test (for example,
TestKafkaProtocolParsing/CO-RE/TestKafkaFetchRaw/with_TLS/api12/split/many_partitions),
which have been tracked down to being due to the bpf_probe_read_user()
helper returning -EFAULT.  So a minor fault is seen on read even though
the page has just been written to, for whatever reason (possibly related
to page migration or transparent hugepage support).
    
A possible real fix in USM would be to use sleeping uprobes and
bpf_copy_from_user(), but that is only available on recent kernels.
    
To work around this issue, reuse buffers in the transparent proxy
instead of reallocating new ones every time.  Reusing the same buffer
appears to leads the kernel to be less likely to mess with our pages.

With this work around in place, the PR re-enables the test.

### Motivation

https://datadoghq.atlassian.net/browse/USMON-1025

### Additional Notes

No failures seen in CI with three branch (in order to run more times) with this patch and the tests modified to run 10 times:

https://app.datadoghq.com/ci/test-runs?query=test_level%3Atest%20%40git.repository.id%3Agitlab.ddbuild.io%2FDataDog%2Fdatadog-agent%20%40git.branch%3Avincent.whitchurch%2Fflaky-bufferfix%2A%20%40test.name%3A%2AKafka%2A&agg_m=count&agg_m_source=base&agg_t=count&citest_explorer_sort=timestamp%2Cdesc&cols=%40test.status%2Ctimestamp%2C%40test.suite%2C%40test.name%3A581%2C%40duration%3A150%2C%40test.service%3A104%2C%40git.branch&currentTab=overview&eventStack=&fromUser=false&index=citest&start=1719469768968&end=1720074568968&paused=false

### CODE REVIEW

File: pkg/network/tracer/testutil/proxy/unix_transparent_proxy.go [hunk 1]
```diff
@@ -154,16 +154,22 @@ func (p *UnixTransparentProxyServer) handleConnection(unixSocketConn net.Conn) {
...
 			for {
-				buf := make([]byte, 8)
+				buf := big[0:8]
 				_, err := io.ReadFull(unixReader, buf)
 				if err != nil {
 					break
 				}
 				readSize := binary.BigEndian.Uint64(buf)
 
 				if readSize != 0 {
```
Comment (guyarb) [inline pkg/network/tracer/testutil/proxy/unix_transparent_proxy.go:171]: edge case - verify readSize is not bigger than 1MB (the size of `big`)

Comment (guyarb) [reply pkg/network/tracer/testutil/proxy/unix_transparent_proxy.go:171]: same for line 194

**Review by guyarb — APPROVED** [review]
small edge-case to handle

File: pkg/network/tracer/testutil/proxy/unix_transparent_proxy.go (hunk 1 above)
Comment (vitkyrka) [reply pkg/network/tracer/testutil/proxy/unix_transparent_proxy.go:171]: Done.

**Pull request merged**